addons/pos_viva_com/static/src/app/hooks/use_viva_app.js

```diff
@@ -0,0 +1,143 @@
+import { usePos } from "@point_of_sale/app/hooks/pos_hook";
+import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { useService } from "@web/core/utils/hooks";
+import { roundPrecision } from "@web/core/utils/numbers";
+import { _t } from "@web/core/l10n/translation";
+import { uuidv4 } from "@point_of_sale/utils";
+
+/**
+ * This hook is used to handle the Viva Wallet app integration in the POS.
+ * API documentation: https://developer.viva.com/apis-for-point-of-sale/card-terminal-apps/android-app/sale/
+ */
+export const useVivaApp = (validateCallback) => {
+    const pos = usePos();
+    const dialog = useService("dialog");
+    const order = pos.getOrder();
+
+    /**
+     * This function determines if the payment method is a Viva Wallet and
+     * if the device should use the Viva Wallet app to process the payment.
+     *
+     * @param {Object} pm - The payment method object.
+     */
+    const use = (pm) => {
+        const isAndroid = navigator.userAgent.toLowerCase().includes("android");
+        return isAndroid && pm.use_payment_terminal === "viva_com";
+    };
+
+    /**
+     * Since we have a new router, the vivawallet callback URL will directly
+     * target the receipt screen, so we need to check if the URL contains
+     * the result of the payment and if it does, we need to process it
+     * and update the payment line accordingly.
+     */
+    const process = async () => {
+        const urlParams = new URLSearchParams(window.location.search);
+        const status = urlParams.get("status");
+        const action = urlParams.get("action");
+        const ref = urlParams.get("referenceNumber");
+        const bankId = urlParams.get("bankId");
```

Comment:
this is weird. When would the payment screen have these search params? The callback url of the vivapayclient is this: `${pos.session._base_url}/pos/ui/${pos.config.id}/payment/${order.uuid}`. It doesn't match any of the above search params.

Reply:
These parameters are added by vivaapp